app/controllers/audits_controller.rb

```diff
@@ -0,0 +1,113 @@
+class AuditsController < ApplicationController
+  before_action :check_if_organization_admin
+  before_action :set_audit, only: %i(show edit update destroy finalize)
+
+  def index
+    @selected_location = filter_params[:at_location]
+    @audits = current_organization.audits.filter(filter_params)
+    @storage_locations = Audit.storage_locations_audited_for(current_organization).uniq
+  end
+
+  def show
+    @inventory_items = @audit.storage_location.inventory_items
+  end
+
+  def edit
+    redirect_to audits_path unless @audit&.in_progress?
+    @storage_locations = current_organization.storage_locations
```

Comment:
Just a thought (please push back if you think otherwise) -- If the user is editing an existing Audit, then the inventory they've been tracking has been picked from a storage location -- changing the storage location would (should?) invalidate any entries already made.

I *think* if you're editing an existing `Audit`, this should probably just set the `@storage_location` to whatever was picked. In the interest of simplicity (so that you don't have to add a bunch of conditionals to the view), you could just set `@storage_locations` to a single-element array containing the storage location chosen (ie. `[@audit.storage_location]`)